app/grandchallenge/serving/views.py

```diff
@@ -74,3 +75,30 @@ def serve_submissions(request, *, submission_pk, **_):
         )
 
     raise PermissionDenied
+
+
+def serve_component_interface_value(
+    request, *, component_interface_value_pk, **_
+):
+    try:
+        user, _ = TokenAuthentication().authenticate(request)
+    except (AuthenticationFailed, TypeError):
+        user = request.user
+
+    try:
+        # output should only be connected to a single job; throw error if not?
+        civ = (
+            ComponentInterfaceValue.objects.filter(
+                pk=component_interface_value_pk
+            )
+            .prefetch_related("algorithms_jobs_as_output")
+            .first()
+        )
+
+    except ComponentInterfaceValue.DoesNotExist:
+        raise Http404("No ComponentInterfaceValue found.")
+
+    if user.has_perm("view_job", civ.algorithms_jobs_as_output.first()):
```

Comment:
We can squash the multiple jobs problem by re-writing this a little. Something like:

```
get_objects_for_user(user=user, perms="algorithms.view_job").filter(outputs__pk=component_interface_value_pk).exists()
```

I've not tested this so we would need to make sure that really works, but don't spend too much time on that if it doesn't.